public/scripts/s3-helpers.js

```diff
@@ -0,0 +1,49 @@
+function saveFileToLocalStorage(file, fileSource) {
+  const reader = new FileReader();
+  reader.addEventListener('load', () => {
+    localStorage.setItem(`toSaveFrom${fileSource}`, reader.result);
+    localStorage.setItem(`${fileSource}fileName`, reader.name);
+  }, false);
+
+  if (file) {
+    reader.readAsDataURL(file);
+  }
+}
+
+function retrieveFile(fileSource) {
+  const file = localStorage.getItem(`toSaveFrom${fileSource}`);
+  const fileName = localStorage.getItem(`${fileSource}fileName`);
+  if (file && fileName) {
+    const retrievedFile = dataURLtoFile(image, fileName);
+    return retrievedFile;
+  }
+
+  return null;
+}
+
+function addTagAndHeadingToDB(tag, heading, imageId) {
+  $.ajax({
+    method: 'PUT',
+    url: 'memory-input-photo',
+    data: { tag, heading, imageId },
+    success: () => { uploadPending = false; },
```

Comment:
No global scope please. Where has this come from?

Not sure what is going on / how I missed this, but you want to be writing your code with require / module.exports.

Everything in the top level of this file is declared in global scope. Further, this code is not being passed though babel (you browserify script does this).

That means your arrow functions are not being converted into es5 and will break some browsers